### PR TITLE
Phytium PWM support

### PR DESCRIPTION
**Pull request by xuyan213**
Generic PWM framework driver for the PWM controller found on Phytium SoCs.

### CODE REVIEW

File: drivers/pwm/pwm-phytium.c [hunk 1]
```diff
@@ -0,0 +1,590 @@
...
+		if (priv->parameter[i].cntmod > 1 ||
+			priv->parameter[i].dutymod > 1 ||
+			priv->parameter[i].div > 4096 ||
+			priv->parameter[i].dbpolarity > 3)
+			return -EINVAL;
+	}
+
+	return phytium_pwm_set_parameter(priv);
+}
+static int pwm_phytium_probe(struct platform_device *pdev)
```
Comment (Avenger-285714) [inline drivers/pwm/pwm-phytium.c:447]: 少了个空行嘛

File: drivers/pwm/sysfs.c [hunk 2]
```diff
@@ -260,7 +260,7 @@ static int pwm_export_child(struct device *parent, struct pwm_device *pwm)
 	export->child.parent = parent;
 	export->child.devt = MKDEV(0, 0);
 	export->child.groups = pwm_groups;
-	dev_set_name(&export->child, "pwm%u", pwm->hwpwm);
+	dev_set_name(&export->child, "pwm%u", pwm->pwm);
```
Comment (Avenger-285714) [inline drivers/pwm/sysfs.c]: 这里为什么要改？

Comment (opsiff) [reply drivers/pwm/sysfs.c]: 这里用CONFIG_ARCH_PHYTIUM宏控制编译实质上会在arm架构上全部使用dev_set_name(&export->child, "pwm%u", pwm->pwm); 此处应该额外在宏控制前提下加read_cpuid_implementor() == ARM_CPU_IMP_PHYTIUM来在ARM上判断飞腾impl吧

**Review by opsiff — CHANGES_REQUESTED** [review]

**Comment by opsiff** [comment]
在 X86 上编译错误 ，之前的ARM架构宏还需要保留
drivers/pwm/sysfs.c: In function ‘pwm_export_child’:
drivers/pwm/sysfs.c:264:13: error: implicit declaration of function ‘read_cpuid_implementor’ [-Werror=implicit-function-declaration]
  264 |         if (read_cpuid_implementor() == ARM_CPU_IMP_PHYTIUM)
      |             ^~~~~~~~~~~~~~~~~~~~~~
drivers/pwm/sysfs.c:264:41: error: ‘ARM_CPU_IMP_PHYTIUM’ undeclared (first use in this function)
  264 |         if (read_cpuid_implementor() == ARM_CPU_IMP_PHYTIUM)
      |                                         ^~~~~~~~~~~~~~~~~~~
drivers/pwm/sysfs.c:264:41: note: each undeclared identifier is reported only once for each function it appears in
cc1: all warnings being treated as errors

**Comment by opsiff** [comment]
剩余审查意见: read_cpuid_implementor方法在一个cpu中不同core有除了飞腾implementor时有其他implementor时可能失效，当前暂时无此种情况。

**Pull request merged**